pulpcore/tasking/tasks.py

```diff
@@ -230,6 +216,32 @@ def enqueue_with_reservation(
     )
 
 
+def _validate_and_get_resources_as_set(resources):
+    resource_set = set()
+    for r in resources:
+        if isinstance(r, str):
+            resource_set.add(r)
+        elif isinstance(r, Model):
+            resource_set.add(util.get_url(r))
+        else:
+            raise ValueError(_("Must be (str|Model)"))
+    return resource_set
+
+
+class UUIDEncoder(json.JSONEncoder):
+    def default(self, obj):
+        deprecation_logger.warn(
+            _(
+                "The argument {obj} is of type {type}, which is not JSON serializable. The use "
+                "of non JSON serializable objects for `args` and `kwargs` to tasks is "
+                "deprecated as of pulpcore==3.12. See the traceback below for more info."
+            ).format(obj=obj, type=type(obj))
+        )
+        if isinstance(obj, uuid.UUID):
+            return str(obj)
+        return json.JSONEncoder.default(self, obj)
```

Comment:
```suggestion
    def default(self, obj):
        if isinstance(obj, uuid.UUID):
            deprecation_logger.warn(
                _(
                    "The argument {obj} is of type {type}, which is not JSON serializable. The use "
                    "of non JSON serializable objects for `args` and `kwargs` to tasks is "
                    "deprecated as of pulpcore==3.12. See the traceback below for more info."
                ).format(obj=obj, type=type(obj))
            )
            return str(obj)
        return json.JSONEncoder.default(self, obj)
```

Also we could just not warn about it and keep the conversion of UUIDs as a convenience to our plugin writers.